src/Altinn.Platform/Altinn.Platform.Storage/IntegrationTest/InstancesQueryAndHALTests.cs

```diff
@@ -0,0 +1,446 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using Altinn.Platform.Storage.Client;
+using Altinn.Platform.Storage.Helpers;
+using Altinn.Platform.Storage.IntegrationTest.Fixtures;
+using Altinn.Platform.Storage.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Storage.Interface.Models;
+using Xunit;
+
+namespace Altinn.Platform.Storage.IntegrationTest
+{
+    /// <summary>
+    ///  Tests dataservice REST api.
+    /// </summary>
+    public class InstancesQueryAndHALTests : IClassFixture<PlatformStorageFixture>, IClassFixture<DatabaseFixture>
+    {
+        private readonly PlatformStorageFixture fixture;
+        private readonly InstanceClient storageClient;
+        private readonly string testOrg = "tdd";
+        private readonly string testAppId = "tdd/m1000";
+        private readonly int testInstanceOwnerId = 1001;
+        private readonly string versionPrefix = "/storage/api/v1";       
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InstanceStorageTests"/> class.
+        /// </summary>
+        /// <param name="fixture">the fixture object which talks to the SUT (System Under Test)</param>
+        public InstancesQueryAndHALTests(PlatformStorageFixture fixture)
+        {
+            this.fixture = fixture;
+            this.storageClient = new InstanceClient(this.fixture.CreateClient());
+
+            CreateTestApplication(testAppId);
+
+            // collectionFixture.LoadData(testAppId, this.storageClient);           
+        }        
+
+        /// <summary>
+        ///  Checks that multiple instances can be returned with query param.
+        /// </summary>
+        [Fact]
+        public async void QueryProcessCurrentTaskSubmit()
+        {
+            HttpClient client = fixture.CreateClient();
+
+            string url = $"{versionPrefix}/instances?appId={testAppId}&size=100&process.currentTask=Submit_1";
+
+            client.DefaultRequestHeaders.Add("Accept", "application/hal+json");
+
+            HttpResponseMessage response = await client.GetAsync(url);
+            response.EnsureSuccessStatusCode();
+
+            string jsonString = await response.Content.ReadAsStringAsync();
+            JObject jsonObject = JObject.Parse(jsonString);
+            int totalHits = jsonObject["totalHits"].Value<int>();
+
+            Assert.Equal(200, totalHits);
+        }
+
+        /// <summary>
+        ///  Checks that multiple instances can be returned with query param.
+        /// </summary>
+        [Fact]
+        public async void QueryProcessVisibleDateTimeGt()
+        {
+            HttpClient client = fixture.CreateClient();
+
+            string url = $"{versionPrefix}/instances?appId={testAppId}&size=100&visibleDateTime=gt:2019-05-01";
```

Comment:
What does "gt" and "lt" stand for?

Reply:
Have commented in code and it is documented in altinn studio docs